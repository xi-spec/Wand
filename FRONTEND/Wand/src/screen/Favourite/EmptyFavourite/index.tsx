import React from 'react';
import { View, Text, Image, TouchableOpacity } from 'react-native';
import styles from './styles';
import { IMAGES } from '../../../constants/images';
import { FAVOURITE_TEXT } from '../../../constants/screenTexts';

export default function EmptyFavourite ({ navigation }:{navigation:
  {navigate:Function, reset:Function}}) {
  return (
        <View style={styles.container}>
          <Image style={styles.image}source={{ uri: IMAGES.emptyFavourite }}></Image>
        <Text style={styles.text}>{FAVOURITE_TEXT.EMPTY_FAVOURITE_TITLE}</Text>
        <Text style={styles.text}>{FAVOURITE_TEXT.EMPTY_FAVOURITE_SUBTITLE}</Text>

        <View style={styles.buttonsContainer}>

        <TouchableOpacity
        testID='shopNow'
        style={{ ...styles.button, ...styles.colorButton }}
        onPress={() => navigation.reset({
          index: 0,
          routes: [{ name: 'Home' }]
        })}>
            <Text style={styles.buttonText}>SHOP NOW</Text>
          </TouchableOpacity>

          <TouchableOpacity
          testID="viewCart"
           style={{ ...styles.button, ...styles.whiteButton }}
           onPress={() => navigation.navigate('Cart')}>
            <Text style={{ ...styles.buttonText, ...styles.blackText }}>VIEW CART</Text>
          </TouchableOpacity>
        </View>

      </View>

  );
}
