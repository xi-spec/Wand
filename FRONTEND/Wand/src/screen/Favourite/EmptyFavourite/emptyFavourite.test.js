import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import EmptyFavourite from './index';

describe('Give a EmptyFavourite component', () => {
  let navigation;

  beforeEach(() => {
    navigation = {
      navigate: jest.fn(),
      reset: jest.fn()
    };
  });
  describe('When is rendered', () => {
    test('Then is rendered correct', () => {
      const tree = render(<EmptyFavourite />);

      expect(tree).toMatchSnapshot();
    });
    describe('When is pressed a shopNow button or viewCart button', () => {
      test('Then navigation.navigate or reset is invoked', () => {
        const { getByTestId } = render(<EmptyFavourite navigation={navigation} />);

        fireEvent.press(getByTestId('shopNow'));
        expect(navigation.reset).toHaveBeenCalled();

        fireEvent.press(getByTestId('viewCart'));
        expect(navigation.navigate).toHaveBeenCalled();
      });
    });
  });
});
