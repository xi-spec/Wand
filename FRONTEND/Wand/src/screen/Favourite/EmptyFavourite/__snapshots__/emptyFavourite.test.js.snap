// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Give a EmptyFavourite component When is rendered Then is rendered correct 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingTop": 266.8,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://trello-attachments.s3.amazonaws.com/603be1b68ce9c654d30ae543/60506d7b3b22568450a0dc85/9d436ab3753190846db49e3ce1cf6950/final1-1024x1024.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 50,
        "height": 100,
        "width": 100,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "RalewayRegular",
        "fontSize": 14,
        "marginHorizontal": 25,
        "paddingVertical": 20,
      }
    }
  >
    YOUR WISHLIST IS EMPTY
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "RalewayRegular",
        "fontSize": 14,
        "marginHorizontal": 25,
        "paddingVertical": 20,
      }
    }
  >
    Check if there any products in your basket and snatch them up before they're gone
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "bottom": 266.8,
        "height": 80,
        "position": "absolute",
        "width": "90%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2E3192",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
      testID="shopNow"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        SHOP NOW
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
      testID="viewCart"
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        VIEW CART
      </Text>
    </View>
  </View>
</View>
`;
